### DISCUSSION TITLE
Porting extensions from other Scratch mods

### DISCUSSION THREAD

**Discussion by BlueDome77** [TurboWarp/extensions#619] [Q&A] [answered]
I have found the source code for a Spotify extension at [scratch.machinelearningforkids.co.uk](https://scratch.machinelearningforkids.co.uk/) that I want to use in **Turbowarp** but I need help on how to modify the code to convert it from a **Module** to an **Extension**.

I can provide the code upon request.

**Comment by GarboMuffin** [comment, marked as answer]
if you intend to submit it here, you need to show that the source code is available under an open source license

unless they've changed much, it should be pretty simple, requiring at least:

1. remove all the require()
2. replace BlockType -> Scratch.BlockType, ArgumentType -> Scratch.ArgumentType
3. replace module.exports = ... with Scratch.extensions.register(new ...)
4. replace this.runtime -> Scratch.vm.runtime
5. wrap in IIFE
